boxsdk/auth/jwt_auth.py

```diff
@@ -209,6 +218,44 @@ def _auth_with_jwt(self, sub, sub_type):
             data['box_device_name'] = self._box_device_name
         return self.send_token_request(data, access_token=None, expect_refresh_token=False)[0]
 
+    def _auth_with_jwt(self, sub, sub_type):
+        """
+        Auth with JWT.
+        If authorization fails because the expiration time is out of sync with the Box servers,
+        retry using the time returned in the error response.
+        Pass an enterprise ID to get an enterprise token (which can be used to provision/deprovision users),
+        or a user ID to get a user token.
+
+        :param sub:
+            The enterprise ID or user ID to auth.
+        :type sub:
+            `unicode`
+        :param sub_type:
+            Either 'enterprise' or 'user'
+        :type sub_type:
+            `unicode`
+        :return:
+            The access token for the enterprise or app user.
+        :rtype:
+            `unicode`
+        """
+        try:
+            return self._construct_and_send_jwt_auth(sub, sub_type)
+        except BoxOAuthException as ex:
+            error_response = ex.network_response
+            try:
+                status_code = error_response.status_code
+                json_response = error_response.json()
+                error_description = json_response['error_description']
+                box_date_header = error_response.headers['date']
+                box_datetime = datetime.strptime(box_date_header, '%a, %d %b %Y %H:%M:%S %Z')
+            except Exception:  # pylint:disable=broad-except
+                pass
+            else:
+                if status_code == 400 and 'exp' in error_description:
```

Comment:
Can we also test for `json_response['error'] == 'invalid_grant'`? I don't like solely depending on a human-readable string.